src/actions/PermissionsActions.js

```diff
@@ -83,12 +98,35 @@ export const promptForGalleryPermission = () => {
         let {status} = await Permissions.askAsync(Permissions.CAMERA_ROLL);
 
         if (status === GRANTED) {
-            getImageFromGallery(dispatch);
+            getImage(dispatch, imageSourceTypes.CAMERA_ROLL);
         } else {
-           return dispatch({
+            return dispatch({
                 type: CAMERA_ROLL_PERMISSION_UPDATED,
                 payload: {cameraRollPermission: status}
             })
         }
     };
+};
+
+export const launchCamera = () => {
+    return async (dispatch) => {
+        getImage(dispatch, imageSourceTypes.CAMERA);
+    };
+};
+
+export const promptForCameraPermission = () => {
+
+    return async (dispatch) => {
+        let {status} = await Permissions.askAsync(Permissions.CAMERA);
+
+        if (status === GRANTED) {
+            getImage(dispatch, imageSourceTypes.CAMERA);
+        } else {
+            return dispatch({
+                type: CAMERA_PERMISSION_UPDATED,
+                payload: {cameraPermission: status}
+            })
+        }
+    };
```

Comment:
Above is the same pattern we used in `promptForCameraRollPermission` where we check for permission, and if it's granted we call `getImage`